docs/use/formatting_outputs.md

```diff
@@ -0,0 +1,218 @@
+pandas.DataFrame({"column 1": [1, 2, 3]})
+```
+````
+
+```{code-cell} ipython3
+:tags: [remove-input,remove-stdout,remove-stderr]
+
+import pandas, sys
+print("this is some stdout")
+print("this is some stderr", file=sys.stderr)
+# but what I really want to show is:
+pandas.DataFrame({"column 1": [1, 2, 3]})
+```
+
+(use/format/images)=
+## Images
+
+With the default renderer, for any image types output by the code, we can apply formatting *via* cell metadata.
+The keys should be placed under `myst`, then for the image we can apply all the variables of the standard [image directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#image):
+
+- **width**: length or percentage (%) of the current line width
+- **height**: length
+- **scale**: integer percentage (the "%" symbol is optional)
+- **align**: "top", "middle", "bottom", "left", "center", or "right"
+- **classes**: space separated strings
+- **alt**: string
+
+Units of length are: 'em', 'ex', 'px', 'in', 'cm', 'mm', 'pt', 'pc'
+
+We can also set a caption (which is rendered as [CommonMark](https://commonmark.org/)) and name, by which to reference the figure:
+
+````md
+```{code-cell} ipython3
+---
+myst:
+  image:
+    width: 200px
+    alt: fun-fish
+    classes: shadow bg-primary
+  figure:
+    caption: |
+      Hey everyone its **party** time!
+    name: fun-fish
+---
+from IPython.display import Image
+Image("images/fun-fish.png")
+```
+````
+
+```{code-cell} ipython3
+---
+myst:
+  image:
+    width: 200px
+    alt: fun-fish
+    classes: shadow bg-primary
+  figure:
+    caption: |
+      Hey everyone its **party** time!
+    name: fun-fish
+---
+from IPython.display import Image
+Image("images/fun-fish.png")
+```
+
+Now we can link to the image: [swim to the fish](fun-fish)
+
+(use/format/markdown)=
+## Markdown
+
+Markdown output is parsed by MyST-Parser, currently with the configuration set to `myst_commonmark_only=True` (see [MyST configuration options](myst:intro/config-options)).
+
+The parsed Markdown is integrated into the wider documentation, and so it is possible, for example, to include internal references:
+
+```{code-cell} ipython3
+from IPython.display import display, Markdown
+display(Markdown('**_some_ markdown** and an [internal reference](use/format/markdown)!'))
+```
+
+and even internal images can be rendered!
+
+```{code-cell} ipython3
+display(Markdown('![figure](../_static/logo.png)'))
+```
+
+(use/format/ansi)=
+## ANSI Outputs
+
+By default, the standard output/error streams and text/plain MIME outputs may contain ANSI escape sequences to change the text and background colors.
+
+```{code-cell} ipython3
+import sys
+print("BEWARE: \x1b[1;33;41mugly colors\x1b[m!", file=sys.stderr)
+print("AB\x1b[43mCD\x1b[35mEF\x1b[1mGH\x1b[4mIJ\x1b[7m"
+      "KL\x1b[49mMN\x1b[39mOP\x1b[22mQR\x1b[24mST\x1b[27mUV")
+```
+
+This uses the built-in {py:class}`~myst_nb.ansi_lexer.AnsiColorLexer` [pygments lexer](https://pygments.org/).
```

Comment:
@chrisjsewell is there a reason we don't just import this from `pygments`. Avoiding a hard requirement?

Reply:
Because its not part of pygments, there is no built-in ANSI lexer, this is a bespoke one

Reply:
It appears to work well for the 8-bit ansi though, perhaps some finessing of the CSS.
Then, as it says in the docs, we can get round to the adding 256-bit support as and when necessary

Reply:
Oh OK -- sorry had assumed it was adapted from `pygments` re: header comment. Makes sense.

Reply:
Is this something we should contribute upstream to `pygments` at some point?

Reply:
Yeh possibly; just wanted to get everything working first, and then we can look to tidy up